cadc-dali/src/test/java/ca/nrc/cadc/dali/ShapeTest.java

```diff
@@ -0,0 +1,164 @@
+    @Test
+    public void testCircle() throws Exception {
+        Circle c = new Circle(new Point(10.0, 20.0), 1.0);
+        Assert.assertEquals(10.0, c.getCenter().getLongitude(), 1.0e-6);
+        Assert.assertEquals(20.0, c.getCenter().getLatitude(), 1.0e-6);
+        Assert.assertEquals(1.0, c.getRadius(), 1.0e-6);
+        
+        Assert.assertEquals(Math.PI, c.getArea(), 1.0e-6);
+        
+        try {
+            Circle rad = new Circle(new Point(10.0, 20.0), -1.0);
+            Assert.fail("expected IllegalArgumentException, got: " + rad);
+        } catch (IllegalArgumentException expected) {
+            log.info("caught expected: " + expected);
+        }
+    }
+
+    @Test
+    public void testRange() throws Exception {
+        Range r = new Range(new DoubleInterval(10.0, 11.0), new DoubleInterval(-0.5, 0.5));
+        Assert.assertEquals(10.5, r.getCenter().getLongitude(), 1.0e-6);
+        Assert.assertEquals(0.0, r.getCenter().getLatitude(), 1.0e-6);
+
+        // area at equator
+        Assert.assertEquals(1.0, r.getArea(), 1.0e-2);
+        
+        // area at high lat
+        r = new Range(new DoubleInterval(10.0, 11.0), new DoubleInterval(60.0, 61.0));
+        Assert.assertEquals(0.5, r.getArea(), 1.0e-2);
+        
+        r = new Range(new DoubleInterval(10.0, 11.0), new DoubleInterval(-61.0, -60.0));
+        Assert.assertEquals(0.5, r.getArea(), 1.0e-2);
+        
+        try {
+            Range ir = new Range(new DoubleInterval(-1.0, 1.0), new DoubleInterval(12.0, 13.0));
+            Assert.fail("expected IllegalArgumentException, got: " + ir);
+        } catch (IllegalArgumentException expected) {
+            log.info("caught expected: " + expected);
+        }
+        
+        try {
+            Range ir = new Range(new DoubleInterval(1.0, 2.0), new DoubleInterval(91.0, 93.0));
+            Assert.fail("expected IllegalArgumentException, got: " + ir);
+        } catch (IllegalArgumentException expected) {
+            log.info("caught expected: " + expected);
+        }
+        
+        try {
+            Range ir = new Range(new DoubleInterval(1.0, 2.0), new DoubleInterval(-91.0, -90.0));
+            Assert.fail("expected IllegalArgumentException, got: " + ir);
+        } catch (IllegalArgumentException expected) {
+            log.info("caught expected: " + expected);
+        }
+    }
+    
+    @Test
+    public void testCircleToPolygonApproximatiom() {
+        try {
+            Circle c = new Circle(new Point(12.0, 34.0), 1.0);
+            double ca = c.getArea();
+
+            for (int i = 4; i < 21; i++) {
+                Polygon poly = Circle.generatePolygonApproximation(c, i);
+                double pa = poly.getArea();
+                double da = pa / ca;
+                log.info("n=" + i + " poly: " + pa + " (" + da + ")");
+            }
+            log.info("circle: " + ca);
+        } catch (Exception unexpected) {
+            log.error("unexpected exception", unexpected);
```

Comment:
If you let the exception pass, it will be reported as an exception which should be unexpected, no?

Reply:
in some cases, this style prints more detail than if the exception is just thrown